pd3/src/main/java/de/dema/pd3/controller/UserController.java

```diff
@@ -190,5 +194,46 @@ public boolean changeChatroomName(@RequestParam("roomId") Long roomId, @RequestP
     	
     	return userService.renameChatroom(id, roomId, name);
     }
+
+    @GetMapping("/public/forgot-password")
+    public String forgotPassword() {
+    	return "public/forgot-password";
+    }
+    
+    @PostMapping("/public/forgot-password")
+    public String forgotPassword(Model model, @RequestParam("email") String email, HttpServletRequest request) {
+    	email = email.trim();
+    	if (EMAIL_PATTERN.matcher(email).matches()) {

```

Comment:
Hier mache ich die Validierung selber, weil die Validierungs-Annotationen mit RequestParams nicht funktioniert. Das ist so aber nicht ideal, weil wir nun zwei potentiell verschiedene Email-Validierungen vornehmen. Zwei Möglichkeiten ergeben sich meines Erachtens:
1. Eine DTO-Klasse mit nur einem Feld namens "email" erzeugen, das die Validierungs-Annotationen bekommt und das dann anstelle des RequestParams verwenden
2. Die hier verwendete Regex als Konstante speichern und diese bei der Verwendung der Email-Annotation immer mit angeben

Was meinst Du? Im Moment plädiere ich für Nummer 1. In dem Fall muss man die Anzeige der Validierungsinformation im Frontend noch entsprechend anpassen.